treeherder/services/pulse/consumers.py

```diff
@@ -186,8 +187,39 @@ def on_message(self, body, message):
         message.ack()
 
 
+class JointConsumer(PulseConsumer):
+    queue_suffix = env("PULSE_QUEUE_NAME", default="queue")
+
+    def bindings(self):
+        rv = []
+        if self.source.get('hgmo'):
+            rv += HGMO_PUSH_BINDINGS
+        elif self.source.get('github'):
+            rv += GITHUB_PUSH_BINDINGS
+        else:
+            return TASKCLUSTER_TASK_BINDINGS
```

Comment:
This whole block does not have the correct behaviour.

Look at this:
```python
        if self.source.get('hgmo'):
            rv += HGMO_PUSH_BINDINGS
        if self.source.get('github'):
            rv += GITHUB_PUSH_BINDINGS
```
It's an array we append to. If you have an `elif` instead of an `if` it won't work.

The behaviour intended is controlling if to listen for pushes from Hg, listening for pushes from Github or listening for tasks.

This probably requires adding one more property to control reading tasks or not (see I add a 3rd property to the dictionary below:
`"github": True, "hgmo": True, "tasks": True`

Reply:
That's why it was falling short of some pulses when I was testing! Understood the issue!